src/OutputWriters/checkpointer.jl

```diff
@@ -201,8 +201,13 @@ function set!(model, filepath::AbstractString)
         # Validate the grid
         checkpointed_grid = file["grid"]
 
-        model.grid == checkpointed_grid ||
-            error("The grid associated with $filepath and model.grid are not the same!")
+	if model.grid isa ImmersedBoundaryGrid
+            model.grid.grid == checkpointed_grid || 
```

Comment:
🤦 

We should change this property name to `underlying_grid` or something (future PR...)

Reply:
yeah that would be usefull